### PR TITLE
Validate email address

### LINKED ISSUE #1308: User is able to pass any string as an email address in Form metadata
#### Problem description
We are able to pass whatever we want as an email address in Form metadata.

#### Steps to reproduce the problem
1. Go to `General Settings -> User and device identity -> Form metadata`
2. Click on `Email address`

#### Expected behavior
Entered value should be validated

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #1308 
I decided to use really simple regex because I tried some complex (and even really complex regex) and there is no perfect one. As it's not really important I think we can use this simple one (it's better than adding complex regex that may block some valid emails).

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/preferences/FormMetadataFragment.java [hunk 1]
```diff
@@ -80,8 +82,16 @@ private void initPrefFromProp(PropertyManager propertyManager,
         return new Preference.OnPreferenceChangeListener() {
             @Override
             public boolean onPreferenceChange(Preference preference, Object newValue) {
-                EditTextPreference changedTextPref = (EditTextPreference) preference;
                 String newValueString = newValue.toString();
+
+                if (KEY_METADATA_EMAIL.equals(key)) {
+                    if (!isEmailAddressValid(newValueString)) {
+                        ToastUtils.showLongToast(getString(R.string.invalid_email_address));
```
Comment (shobhitagarwal1612) [inline collect_app/src/main/java/org/odk/collect/android/preferences/FormMetadataFragment.java]: `getString(R.string.invalid_email_address)` can be replaced with `R.string.invalid_email_address`

**Comment by grzesiek2010** [comment]
I think it would be good to add test case so I'll do that when I'm free.

**Comment by mmarciniak90** [comment]
Tested on Android 4.1, 4.2, 4.4, 5.1, 6.0, 7.0
Fix is not perfect because it accepts invalid email addresses like eg: 
* a@a@a@a.com (only one @ is allowed outside quotation marks)
* a"b(c)d,e:f;g<h>i[j\k]l@example.com (none of the special characters in this local-part are allowed outside quotation marks)
* john..doe@example.com (double dot before @)

and does not accept valid:
* user@localserver

(Based on https://en.wikipedia.org/wiki/Email_address)
But as @grzesiek2010 mentioned, he used really simple regex.

User is additionally able to remove email address and leaves the field empty.
User sees toast when user puts invalid email address:
screenshot_20170810-114450

**Comment by lognaturel** [comment]
My sense is that this check doesn't add a lot of value to the user and potentially causes frustration! This might be a good thing for users to just do on the server side based on the email address formats they expect. @yanokwa?

**Comment by lognaturel** [comment]
As we discussed briefly, let's just require an `@`.

**Comment by yanokwa** [comment]
I'd vote for a slight improvement, say `\S+@\S+`

**Comment by grzesiek2010** [comment]
@lognaturel I decided to use `.+@.+` instead of `\S+@\S+` because white spaces are allowed eg.: 
`" "@example.org`
 and additionally I set `maxLength` for  the [EditText](https://github.com/opendatakit/collect/pull/1299/files#diff-d85822923db337c3bcff225235efbe29R15) to 254 as it's a max valid length according to https://en.wikipedia.org/wiki/Email_address

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Wow, I didn't know that about spaces. Sigh.

**Comment by akshay-ap** [comment]
I had thought of following standard `RFC5322` / `RFC2822`/`RFC822` for email validation.
Here is the standard specification :
https://www.ietf.org/rfc/rfc5322.txt
Although, in `Collect` case such stringent validation is not needed and doesn't add make big  difference I think.

**Pull request merged**